sample-operators/tomcat-operator/src/main/java/io/javaoperatorsdk/operator/sample/TomcatReconciler.java

```diff
@@ -2,74 +2,48 @@
 import io.javaoperatorsdk.operator.api.reconciler.Reconciler;
 import io.javaoperatorsdk.operator.api.reconciler.UpdateControl;
-import io.javaoperatorsdk.operator.processing.event.source.EventSource;
-import io.javaoperatorsdk.operator.processing.event.source.informer.InformerEventSource;
-import io.javaoperatorsdk.operator.processing.event.source.informer.Mappers;
 
 import static io.javaoperatorsdk.operator.api.reconciler.Constants.NO_FINALIZER;
 
 /**
  * Runs a specified number of Tomcat app server Pods. It uses a Deployment to create the Pods. Also
  * creates a Service over which the Pods can be accessed.
  */
-@ControllerConfiguration(finalizerName = NO_FINALIZER)
-public class TomcatReconciler implements Reconciler<Tomcat>, EventSourceInitializer<Tomcat> {
+@ControllerConfiguration(
+    finalizerName = NO_FINALIZER,
+    dependents = {
+        @Dependent(resourceType = Deployment.class, type = DeploymentDependentResource.class),
+        @Dependent(resourceType = Service.class, type = ServiceDependentResource.class)
+    })
+public class TomcatReconciler implements Reconciler<Tomcat> {
 
   private final Logger log = LoggerFactory.getLogger(getClass());
 
-  private final KubernetesClient kubernetesClient;
-
-  public TomcatReconciler(KubernetesClient client) {
-    this.kubernetesClient = client;
-  }
-
-  @Override
-  public List<EventSource> prepareEventSources(EventSourceContext<Tomcat> context) {
-    SharedIndexInformer<Deployment> deploymentInformer =
-        kubernetesClient.apps().deployments().inAnyNamespace()
-            .withLabel("app.kubernetes.io/managed-by", "tomcat-operator")
-            .runnableInformer(0);
-
-    return List.of(new InformerEventSource<>(
-        deploymentInformer, Mappers.fromOwnerReference()));
-  }
-
   @Override
   public UpdateControl<Tomcat> reconcile(Tomcat tomcat, Context context) {
-    createOrUpdateDeployment(tomcat);
-    createOrUpdateService(tomcat);
-
-    return context.getSecondaryResource(Deployment.class)
-        .map(deployment -> {
-          Tomcat updatedTomcat =
-              updateTomcatStatus(tomcat, deployment);
-          log.info(
-              "Updating status of Tomcat {} in namespace {} to {} ready replicas",
-              tomcat.getMetadata().getName(),
-              tomcat.getMetadata().getNamespace(),
-              tomcat.getStatus().getReadyReplicas());
-          return UpdateControl.updateStatus(updatedTomcat);
-        })
-        .orElse(UpdateControl.noUpdate());
+    return context.getSecondaryResource(Deployment.class).map(deployment -> {
+      Tomcat updatedTomcat = updateTomcatStatus(tomcat, deployment);
+      log.info(
+          "Updating status of Tomcat {} in namespace {} to {} ready replicas",
+          tomcat.getMetadata().getName(),
+          tomcat.getMetadata().getNamespace(),
+          tomcat.getStatus().getReadyReplicas());
+      return UpdateControl.updateStatus(updatedTomcat);
```

Comment:
Is the `update` method from the `DependentResource` called at this point ? like https://github.com/java-operator-sdk/java-operator-sdk/pull/785/files#diff-06a1ae70b5e55bf828c2120ae4525d9ad22cb82b526c6db644472ad46698f6f9R47
And how does it work when we have more than one secondary resource (like having 3 deployments) ?